pkg/endpoint/builder.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2016-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import (
+	"github.com/cilium/cilium/pkg/buildqueue"
+	"github.com/cilium/cilium/pkg/monitor"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+var (
+	// BuildQueue is the endpoint build queue. It is also used to build
+	// base programs
+	BuildQueue = buildqueue.NewBuildQueue()
+)
+
+type endpointBuild struct {
+	endpoint *Endpoint
+	context  *RegenerationContext
+	owner    Owner
+}
+
+func (e *Endpoint) newEndpointBuild(owner Owner, context *RegenerationContext) *endpointBuild {
+	return &endpointBuild{
+		endpoint: e,
+		owner:    owner,
+		context:  context,
+	}
+}
+
+func (b *endpointBuild) GetUUID() string {
+	return b.endpoint.GetUUID()
+}
+
+func (b *endpointBuild) Build() error {
+	e := b.endpoint
+
+	scopedLog := e.getLogger()
```

Comment:
I need to say that if you add in the `e.GetLogger` the UUID for the build you'll make me super happy! So, ops people can grep for the given UUID and will get all the information. 

Maybe add to `b.Context` a logrus.Entry with the uuid, and e.regenerate feed the logs from there. it will be great!

Reply:
We might as well just make UUID part of the endpoint logger to have unique correlation